### ISSUE TITLE
feat(config): configure RPC timeout for forge script

### ISSUE THREAD

**Issue by kevinhalliday** [foundry-rs/foundry#9303]
### Component

Forge

### Describe the feature you would like

Forge script rpc requests timeout after 45 seconds. You can increase timeout for waiting for tx receipts, but you cannot increase timeout for `eth_sendTransaction` requests. The feature would allow using MPC signers (like fireblocks) w/ unlocked forge scripts.

### Additional context

This is a "reopening" of https://github.com/foundry-rs/foundry/issues/8667.

> **Describe the feature you would like**
> Currently forge script rpc requests timeout after 45 seconds. I believe this is not configurable (let me know if I'm wrong). Being able to specify timeout would be 🤌

> **Additional context**
> We are running forge scripts --unlocked against a proxy rpc that intercepts eth_sendTransaction requests and signs them with fireblocks. The MPC signing takes time, often longer than 45seconds. The scripts timeout pretty consistently.

The fix (https://github.com/foundry-rs/foundry/pull/8669) only increased timeout for waiting for tx receipts. But `eth_sendTransaction` requests still use constant 45 second timeout.

**Comment by grandizzy** [comment]
I think we could reuse the existing `eth_rpc_timeout` config when creating script providers, @zerosnacks @klkvr would this work or could be confusing?

**Comment by klkvr** [comment]
yep, we should make sure to add it to `ScriptArgs` to make it available there

right now scripts are using different code path than cast for creating providers, need to make it aware of timeout https://github.com/foundry-rs/foundry/blob/4817280d96e0e33a2e96cf169770da60514d1764/crates/script/src/broadcast.rs#L258

**Comment by grandizzy** [comment]
:+1: , or maybe better an `--rpc-timeout` in `EvmArgs` along with `--rpc-url` ?

https://github.com/foundry-rs/foundry/blob/4817280d96e0e33a2e96cf169770da60514d1764/crates/common/src/evm.rs#L44-L50

**Comment by zerosnacks** [comment]
We have an `rpc_timeout` in `RpcOpts` here

https://github.com/foundry-rs/foundry/blob/4817280d96e0e33a2e96cf169770da60514d1764/crates/cli/src/opts/ethereum.rs#L44-L50

**Comment by klkvr** [comment]
yeah that's a bit weird because we can't use both `EvmArgs` and `RpcOpts` because of `--rpc-url` conflict, might make sense to try unifying those

**Comment by 10kumaranurag01** [comment]
I'd love to give this a go. Please assign this to me

**Comment by mimisavage** [comment]
Could I be assigned to this?

**Comment by chiscookeke11** [comment]
Would love to tackle this!

**Comment by wengDavo** [comment]
Let me try this one!

**Comment by kevinhalliday** [comment]
light poke here from the author 🙏  having this feature would be really helpful

**Comment by grandizzy** [comment]
considering for v1.1

**Comment by KevinSmall** [comment]
This feature would be helpful for our project also
